src/psbt.cpp

```diff
@@ -66,8 +66,11 @@ bool PartiallySignedTransaction::AddOutput(const CTxOut& txout, const PSBTOutput
 bool PartiallySignedTransaction::GetInputUTXO(CTxOut& utxo, int input_index) const
 {
     PSBTInput input = inputs[input_index];
```

Comment:
nit: `inputs.at(input_index)`

Reply:
Unrelated style changes are unrelated

Reply:
It's not a style change, `at` checks these array indexes "are within bounds" and throws an exception otherwise. There's a bunch of places in this code where we don't use iterators and assume that two arrays are the same size.

Reply:
Throwing an exception here would actually be rude, yes? So it would be better to establish invariants that ensure we are in-bounds, or else add a bounds check ourselves otherwise?

Reply:
> establish invariants that ensure we are in-bounds

Agree, I think it makes no sense to add bound checks here.

In any case, here or followup this could use
```
const_reference operator[]( size_type pos ) const;
```
since `PSBTInput` is not that trivial.

Reply:
That would just mean changing "PSBTInput input" to "const PSBTInput& input"?

Reply:
Yes.